src/NLPModelsIpopt.jl

```diff
@@ -122,6 +122,16 @@ end
     set_callbacks(nlp::AbstractNLPModel)
 
 Return the set of functions needed to instantiate an `IpoptProblem`.
+
+This function creates the callback functions that Ipopt needs to evaluate:
+- the objective function
+- the constraint function (if any)
+- the objective gradient
+- the constraint Jacobian (if any) 
+- the Hessian of the Lagrangian
+
+For information on using callbacks to monitor the optimization process, 
+see the tutorial documentation.
```

Comment:
```suggestion
```
I personally don't think we have to document this function that is not exported.